src/pygeom.c

```diff
@@ -76,6 +76,69 @@ static PyObject *GeometryObject_str(GeometryObject *self)
     return GeometryObject_ToWKT(self, "%s");
 }
 
+static Py_hash_t GeometryObject_hash(GeometryObject *self)
+{
+    void *context = geos_context[0];
+    unsigned char *wkb;
+    size_t size;
+    Py_hash_t x;
+
+    if (self->ptr == NULL) {
+        return -1;
+    }
+    GEOSWKBWriter *writer = GEOSWKBWriter_create_r(context);
+    if (writer == NULL) {
+        return -1;
+    }
+
+    GEOSWKBWriter_setOutputDimension_r(context, writer, 3);
+    GEOSWKBWriter_setIncludeSRID_r(context, writer, 1);
+    wkb = GEOSWKBWriter_write_r(context, writer, self->ptr, &size);
+    GEOSWKBWriter_destroy_r(context, writer);
+    if (wkb == NULL) {
+        return -1;
+    }
+    x = PyHash_GetFuncDef()->hash(wkb, size);
+    if (x == -1) {
+        x = -2;
+    } else {
+        x ^= 374761393UL;  // to make the result distinct from the actual WKB hash //
```

Comment:
is this a random number? Or does it come from somewhere?

Reply:
It is a prime that I stole from some other hash function in the Python source. I guess it could be any number if you use it in a XOR like I did.